backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -114,26 +205,92 @@ public String creatTestCVSForDataHub(String lastEndCreateOn) {
-            this._uploadCSVDocument(apiKey);
+            this.createTestEventCSV(lastEndCreateOn);
+            this.uploadCSVDocument(apiKey);
 
-            return  Map.of(
+            return Map.of(
                     "result", "ok",
                     "lastTimestamp", this._nextTimestamp,
                     "rowsSent", String.valueOf(this._rowCount),
-                    "uploadResultId", this._uploadResultId,
+                    "uploadResultId", this._resultJson,
                     "message", this._warnMessage);
         } catch (RestClientException err) {
             return Map.of("result", "error",
-                    "message","Err: uploading csv to data-hub failed. error='" + err.toString() + "'");
+                    "message", "Err: uploading csv to data-hub failed. error='" + err.toString() + "'");
         } catch (IOException err) {
             return Map.of("result", "error",
                     "messsage", err.toString());
         } catch (NoResultException err) {
             return Map.of("result", "error",
-                    "message","No matching results for the given startupdateby param");
+                    "message", "No matching results for the given startupdateby param");
         }
     }
+
+    @Transactional
+    public void dataHubUploaderTask() {
+
+        // sanity check everything is configured correctly (dev likely will not be)
+        if (!_config.getUploadEnabled()) {
+            LOG.info("DataHubUploaderTask not running because simple-report.data-hub.uploadenabled is false");
+            return;
+        }
+
+        ArrayList<String> msgs = new ArrayList<>();
+        final int MIN_KEY_LENGTH = 24;
+        if (_config.getApiKey().length() < MIN_KEY_LENGTH) {
+            msgs.add("> DataHub API key is not configured.");
+        }
+        if (!_config.getUploadUrl().startsWith("https://")) {
+            msgs.add("> DataHub upload URL is not configured.");
+        }
+        if (!msgs.isEmpty()) {
+            sendSlackChannelMessage("DataHubUploader not run", msgs, true);
+            return;
+        }
+
+        // The start date is the last end date. Can be null for empty database.
+        Date lastTimestamp = getLatestRecordedTimestamp();
+        DataHubUpload newUpload = new DataHubUpload(_config)
+                .setJobState("INIT")
+                .setEarliestRecordedTimestamp(lastTimestamp);
+        _dataHubUploadRepo.save(newUpload);
+
+        try {
+            // end range is back 1 minute, this is to avoid selecting transactions that
+            // may still be rolled back.
+            final Timestamp DATE_ONE_MIN_AGO = Timestamp.from(Instant.now().minus(1, ChronoUnit.MINUTES));
+
+            this.createTestEventCSV(lastTimestamp, DATE_ONE_MIN_AGO);
+            _dataHubUploadRepo.save(newUpload
+                    .setRecordsProcessed(_rowCount)
+                    .setLatestRecordedTimestamp(utcStringToDate(_nextTimestamp)));
```

Comment:
Converting this to a string in order to convert it back to a date seems unnecessary. 🙂 

Also, I would think we should change the state from INIT to IN_PROGRESS or something here?